tests/returntypes/transforms/test_split_tape_new.py

```diff
@@ -0,0 +1,343 @@
+from pennylane.transforms import split_tape
+
+dev = qml.device("default.qubit", wires=4)
+"""Defines the device used for all tests"""
+
+H1 = qml.Hamiltonian([1.5, 1.5], [qml.PauliZ(0) @ qml.PauliZ(1), qml.PauliZ(0) @ qml.PauliZ(1)])
+S1 = qml.s_prod(1.5, qml.prod(qml.PauliZ(0), qml.PauliZ(1)))
+
+"""Defines circuits to be used in queueing/output tests"""
+tape1 = QuantumScript(
+    [qml.PauliX(0)], [qml.expval(H1), qml.expval(S1), qml.expval(S1), qml.state()]
+)
+
+H2 = qml.Hamiltonian(
+    [1, 3, -2, 1, 1],
+    [
+        qml.PauliX(0) @ qml.PauliZ(2),
+        qml.PauliZ(2),
+        qml.PauliX(0),
+        qml.PauliX(2),
+        qml.PauliZ(0) @ qml.PauliX(1),
+    ],
+)
+S2 = qml.op_sum(
+    qml.prod(qml.PauliX(0), qml.PauliZ(2)),
+    qml.s_prod(3, qml.PauliZ(2)),
+    qml.s_prod(-2, qml.PauliX(0)),
+    qml.PauliX(2),
+    qml.prod(qml.PauliZ(0), qml.PauliX(1)),
+)
+tape2 = QuantumScript(
+    [qml.Hadamard(0), qml.Hadamard(1), qml.PauliZ(1), qml.PauliX(2)],
+    [qml.expval(H2), qml.expval(S2), qml.probs(op=qml.PauliZ(0)), qml.expval(S2)],
+)
+
+H3 = 1.5 * qml.PauliZ(0) @ qml.PauliZ(1) + 0.3 * qml.PauliX(1)
+S3 = qml.op_sum(
+    qml.s_prod(1.5, qml.prod(qml.PauliZ(0), qml.PauliZ(1))), qml.s_prod(0.3, qml.PauliX(1))
+)
+
+tape3 = QuantumScript(
+    [qml.PauliX(0)],
+    [
+        qml.expval(H3),
+        qml.expval(S3),
+        qml.probs(wires=1),
+        qml.expval(qml.PauliX(1)),
+        qml.expval(S3),
+        qml.probs(wires=[0, 1]),
+    ],
+)
+H4 = (
+    qml.PauliX(0) @ qml.PauliZ(2)
+    + 3 * qml.PauliZ(2)
+    - 2 * qml.PauliX(0)
+    + qml.PauliZ(2)
+    + qml.PauliZ(2)
+    + qml.PauliZ(0) @ qml.PauliX(1) @ qml.PauliY(2)
+)
+S4 = (
+    qml.prod(qml.PauliX(0), qml.PauliZ(2))
+    + qml.s_prod(3, qml.PauliZ(2))
+    - qml.s_prod(2, qml.PauliX(0))
+    + qml.PauliZ(2)
+    + qml.PauliZ(2)
+    + qml.prod(qml.PauliZ(0), qml.PauliX(1), qml.PauliY(2))
+)
+
+tape4 = QuantumScript(
+    [qml.Hadamard(0), qml.Hadamard(1), qml.PauliZ(1), qml.PauliX(2)],
+    [
+        qml.expval(H4),
+        qml.expval(S4),
+        qml.expval(qml.PauliX(2)),
+        qml.expval(S4),
+        qml.expval(qml.PauliX(2)),
+    ],
+)
+TAPES = [tape1, tape2, tape3, tape4]
+OUTPUTS = [
+    [
+        -3.0,
+        -1.5,
+        -1.5,
+        np.array(
+            [
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                1.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+            ]
+        ),
+    ],
+    [-6, -6, np.array([0.5, 0.5]), -6],
+    [-1.5, -1.5, np.array([1.0, 0.0]), 0.0, -1.5, np.array([0.0, 0.0, 1.0, 0.0])],
+    [-8, -8, 0, -8, 0],
```

Comment:
Very nice testing! Could you try to mix two different Hamiltonian in a single tape, where they do not have commuting observables. I don't think we cover this case with the four tapes.

Reply:
Added two more tapes containing expvals of different hamiltonians and sums!